src/main/java/net/dv8tion/jda/internal/requests/restaction/MessageActionImpl.java

```diff
@@ -480,11 +509,15 @@ protected DataObject getJSON()
             if (nonce != null)
                 obj.put("nonce", nonce);
         }
-        obj.put("tts", tts);
-        if (allowedMentions != null || !mentionableUsers.isEmpty() || !mentionableRoles.isEmpty())
+        if (messageReference != 0)
         {
-            obj.put("allowed_mentions", getAllowedMentionsObj());
+            obj.put("message_reference", DataObject.empty()
```

Comment:
Just tested this. Not sure is that an issue or not, but replying is not working in case of just sending a file without any other message content.

In this case rest action ignores this `payload_json` as it thinks it's empty so `message_reference` never sent to the Discord with a file.

Reply:
This has been fixed